### DISCUSSION TITLE
How to automate publishing storefronts?

### DISCUSSION THREAD

**Discussion by RussellSlayton** [swellstores/community#62] [Help] [answered]
So generally what I want to do is writing a script to automatically publishing storefronts with my own website templates(html and css files) with customized part like brand name or products' names, etc.
After I check the swell developers documentations, I see there is a way using git to publish storefronts.
[https://developers.swell.is/storefronts/update-swell-themes-with-git#push-your-updates-live-with-a-git-push](url)
But in that way I have to make every storefront a github repository, since each one of them have different webpages.
Is there any clever way to work this around or another way to do this?

**Comment by rodrigoddalmeida** [comment]
As of now, you can only have one hosted storefront per Swell store, but you can host storefronts in 3rd-party hosting vendors. Those vendors might support multiple websites per project.

As a reference, you can check out [here](https://developers.swell.is/guides/deploy-horizon-to-vercel) how Horizon can be hosted on Vercel. This process should be similar for different apps/templates you might have.

**Reply by ericingram** [reply]
Another way to look at it: if you stored all the products and content in the same Swell account, then you could build a single application that fetches different content based on the domain (routing would be on the application side). Then you only have one app and one hosting instance instead of one for every storefront.

**Comment by logeshswell** [comment, marked as answer]
Just thinking out loud, Does [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) solve your problem for the time being?

Basically, you will have a parent repo that contains all your html and css files with base configs like brand names or products' names, etc.
You can have an individual repo for each storefront containing the parent repo as the git submodule along with the configs that override the parent repo.
So, whenever you make a change in the base repo, you would also need to pull and update your submodules and push it back to publish your storefront. In this way, you will also get control over individual storefronts when publishing your styling changes.

@RussellSlayton Would this method help you?
